### ISSUE TITLE
MapShed AEU failures (TypeError: unsupported operand type(s) for *: 'NoneType' and 'float')

### ISSUE THREAD

**Issue by mmcfarland** [WikiWatershed/model-my-watershed#1477]
View details in Rollbar: [https://rollbar.com/Azavea/MMW/items/112/](https://rollbar.com/Azavea/MMW/items/112/)

```
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "apps/modeling/tasks.py", line 138, in analyze_animals
    return [animal_population(area_of_interest)]
  File "apps/modeling/calcs.py", line 28, in animal_population
    aeu_for_geom = animal_energy_units(geom)[2]
  File "apps/modeling/mapshed/calcs.py", line 190, in animal_energy_units
    for animal in LIVESTOCK))
  File "apps/modeling/mapshed/calcs.py", line 190, in <genexpr>
    for animal in LIVESTOCK))
TypeError: unsupported operand type(s) for *: 'NoneType' and 'float'
```

**Comment by mmcfarland** [comment]
This issue appears to be related to MapShed jobs on areas that have no AEU data.  We should be able to pass along 0 counts for these.  There may be corresponding conditions because I believe I've seen MapShed run successfully within ubran areas with no AEU.

**Comment by kellyi** [comment]
I believe this was fixed in #1464 but I'm going to double check. In that PR I changed it so that the db calls involved in retrieving AEU will return 0.0 for values of `None`:

https://github.com/WikiWatershed/model-my-watershed/pull/1464/commits/26dfa8cfbf91cbe0db5e386e91a7221d0604e73d#diff-f23f8c6b8254839c70f2d4939c060e7cR176

https://github.com/WikiWatershed/model-my-watershed/pull/1464/commits/4727cac952dd4547e85860534dfb5141def366bb

GWLF-E has also been updated to handle AOIs with 0 counts for AEU: https://github.com/WikiWatershed/gwlf-e/pull/71

**Comment by kellyi** [comment]
Just checked this again with this AOI:

![screen shot 2016-09-26 at 10 12 06 am](https://cloud.githubusercontent.com/assets/4165523/18837424/bacf589e-83d1-11e6-9d67-ee3134031868.png)

Attempting to run MapShed led to this error message which was created in #1464/#1466:

![screen shot 2016-09-26 at 10 12 59 am](https://cloud.githubusercontent.com/assets/4165523/18837458/d4a95936-83d1-11e6-9ea1-b4d80147d190.png)

On staging currently this will generate `Exception(NO_LAND_COVER)` errors like this one --https://rollbar.com/Azavea/MMW/items/116/ -- and we created #1486 to reject those shapes on the frontend before sending to MapShed.

**Comment by kellyi** [comment]
Since this was fixed in #1464 and #1466, I'm going to close this and sub in #1486 to finish the frontend guard.
